### ISSUE TITLE
[False-Negative] Non-optimal regular expression, i flag

### ISSUE THREAD

**Issue by puggan** [kalessil/phpinspectionsea#1187]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Negative |
| Plugin         | Php Inspections (EA Ultimate) v2.0.12 |
| Language level | PHP 5.6 |

Current behaviour (description/screenshot:)
--
```php
$n = preg_replace('/\D/i', $s); // No warning
$n = preg_replace('/[^0-9]/i', $s); // Warning for both i-flag and not using \D
```

Expected behaviour
--
Still warn about the i-flag after replacing '/[^0-9]/i' with '/\D/i'.

Guess its treats '\D' as an 'D', and therefor think it have upercase characters,
if thats is the case maybe all letters that are part of an escape sequense should be ignored when looking for uper/lower-case letters.

Environment details
---
PhpStorm 2018.3.3
Build #PS-183.5153.36, built on January 9, 2019
JRE: 1.8.0_152-release-1343-b26 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.18.0-13-generic

**Comment by kalessil** [comment]
Fixed!
